// angular5-location/src/app/location.module.js
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { MatAutocompleteModule, MatFormFieldModule, MatInputModule, MatOptionModule } from '@angular/material';
import { LocationComponent } from './location.component';
import { HttpClient, HttpClientModule } from '@angular/common/http';
import { HttpModule, JsonpModule } from '@angular/http';
import { ExampleComponent } from '../../examples/example.component';
import { PlacesService } from './services/places.service';
var LocationModule = /** @class */ (function () {
    function LocationModule() {
    }
    LocationModule.decorators = [
        { type: NgModule, args: [{
                    imports: [/*BrowserModule,*/ BrowserAnimationsModule, CommonModule, HttpClientModule, HttpModule, JsonpModule, FormsModule, ReactiveFormsModule, MatAutocompleteModule, MatFormFieldModule, MatInputModule, MatOptionModule],
                    providers: [HttpClient, PlacesService],
                    declarations: [LocationComponent, ExampleComponent],
                    exports: [
                        LocationComponent
                    ],
                    bootstrap: [ExampleComponent]
                },] },
    ];
    return LocationModule;
}());
export { LocationModule };
