// ng7-mat-breadcrumb/esm5/lib/ng7-mat-breadcrumb.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Component } from '@angular/core';
import { ActivatedRoute, Router, NavigationEnd, PRIMARY_OUTLET } from '@angular/router';
import { filter } from 'rxjs/operators';
import { map } from 'rxjs/internal/operators';
var Ng7MatBreadcrumbComponent = /** @class */ (function () {
    function Ng7MatBreadcrumbComponent(activatedRoute, router) {
        this.activatedRoute = activatedRoute;
        this.router = router;
        this.breadcrumb = [];
        this.breadCrumbData();
    }
    /**
     * @return {?}
     */
    Ng7MatBreadcrumbComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    /**
     * @return {?}
     */
    Ng7MatBreadcrumbComponent.prototype.breadCrumbData = /**
     * @return {?}
     */
    function () {
        var _this = this;
        this.router.events
            .pipe(filter((/**
         * @param {?} event
         * @return {?}
         */
        function (event) { return event instanceof NavigationEnd; })))
            .pipe(map((/**
         * @return {?}
         */
        function () { return _this.activatedRoute; })))
            .pipe(map((/**
         * @param {?} route
         * @return {?}
         */
        function (route) {
            while (route.firstChild) {
                route = route.firstChild;
            }
            return route;
        })))
            .pipe(filter((/**
         * @param {?} route
         * @return {?}
         */
        function (route) { return route.outlet === PRIMARY_OUTLET; })))
            .subscribe((/**
         * @param {?} route
         * @return {?}
         */
        function (route) {
            if (route.snapshot.data.breadcrumb) {
                /** @type {?} */
                var breadcrumb = (JSON.parse(JSON.stringify(route.snapshot.data.breadcrumb)));
                breadcrumb.map((/**
                 * @param {?} crumb
                 * @return {?}
                 */
                function (crumb) {
                    var e_1, _a;
                    /** @type {?} */
                    var urlChunks = crumb.url.split('/');
                    try {
                        for (var urlChunks_1 = tslib_1.__values(urlChunks), urlChunks_1_1 = urlChunks_1.next(); !urlChunks_1_1.done; urlChunks_1_1 = urlChunks_1.next()) {
                            var chunk = urlChunks_1_1.value;
                            if (chunk.includes(':')) {
                                /** @type {?} */
                                var paramID = chunk.replace(':', '');
                                /** @type {?} */
                                var routerParamID = route.snapshot.params[paramID];
                                crumb.url = crumb.url.replace(":" + paramID, routerParamID);
                            }
                        }
                    }
                    catch (e_1_1) { e_1 = { error: e_1_1 }; }
                    finally {
                        try {
                            if (urlChunks_1_1 && !urlChunks_1_1.done && (_a = urlChunks_1.return)) _a.call(urlChunks_1);
                        }
                        finally { if (e_1) throw e_1.error; }
                    }
                }));
                _this.breadcrumb = breadcrumb;
            }
            else {
                _this.breadcrumb = [];
            }
        }));
    };
    Ng7MatBreadcrumbComponent.decorators = [
        { type: Component, args: [{
                    selector: 'app-ng7-mat-breadcrumb',
                    template: "<mat-toolbar class=\"mat-bread-crumb\">\n    <mat-list class=\"mat-bread-crumb-list\" *ngIf=\"breadcrumb.length\">\n        <mat-list-item>\n            <span class=\"mat-bread-crumb-spacer\">\n                <span *ngFor=\"let item of breadcrumb; let i = index\">\n                    <a *ngIf=\"item?.url\" [routerLink]=\"item?.url\" class=\"mat-bread-crumb-list-item\">{{ item.label }}</a>\n                    <span *ngIf=\"!item?.url\" class=\"mat-bread-crumb-list-item\">{{ item.label }}</span>\n                    <span *ngIf=\"breadcrumb.length !== i+1\"> / </span>\n                </span>\n            </span>\n        </mat-list-item>\n    </mat-list>\n</mat-toolbar>",
                    styles: [".mat-bread-crumb{height:30px;background-color:#f1efeb}.mat-bread-crumb-list{padding-bottom:10px}.mat-bread-crumb-list-item{font-size:15px}.mat-bread-crumb-spacer{margin-left:-20px}"]
                }] }
    ];
    /** @nocollapse */
    Ng7MatBreadcrumbComponent.ctorParameters = function () { return [
        { type: ActivatedRoute },
        { type: Router }
    ]; };
    return Ng7MatBreadcrumbComponent;
}());
export { Ng7MatBreadcrumbComponent };
if (false) {
    /** @type {?} */
    Ng7MatBreadcrumbComponent.prototype.breadcrumb;
    /**
     * @type {?}
     * @private
     */
    Ng7MatBreadcrumbComponent.prototype.activatedRoute;
    /**
     * @type {?}
     * @private
     */
    Ng7MatBreadcrumbComponent.prototype.router;
}
