// angular5-location/src/app/services/places.service.js
import { Injectable } from '@angular/core';
import { Jsonp } from '@angular/http';
import 'rxjs/add/operator/map';
var PlacesService = /** @class */ (function () {
    function PlacesService(jsonp) {
        this.jsonp = jsonp;
        this.apiList = [
            {
                key: 'country',
                path: 'https://battuta.medunes.net/api/country/all/?key={{token}}'
                //path: 'https://api.geonames.org/countryInfo?username=prakash_geo'
            },
            {
                key: 'state',
                //path: 'http://api.geonames.org/children?geonameId={{geonameId}}&username=prakash_geo'
                path: 'https://battuta.medunes.net/api/region/{{country_code}}/all/?key={{token}}'
            },
            {
                key: 'city',
                path: 'https://battuta.medunes.net/api/city/{{country_code}}/search/?region={{city}}&key={{token}}'
            }
        ];
    }
    PlacesService.prototype.getDataFromCloud = function (apipath) {
        var apiURL = apipath + "&callback=JSONP_CALLBACK";
        return this.jsonp.request(apiURL)
            .map(function (res) {
            return res.json();
        });
    };
    PlacesService.prototype.getApiList = function () {
        return this.apiList;
    };
    PlacesService.decorators = [
        { type: Injectable },
    ];
    /** @nocollapse */
    PlacesService.ctorParameters = function () { return [
        { type: Jsonp, },
    ]; };
    return PlacesService;
}());
export { PlacesService };
